### PR TITLE
Store multiple game logs with time stamp

### LINKED ISSUE #777: Seperate log file per game
image

### PR DESCRIPTION
**Pull request by Sheikah45**
Fixes #777

Stores up to 10 game logs in the log folder with the time stamp

### CODE REVIEW

File: src/main/java/com/faforever/client/preferences/PreferencesService.java [hunk 1]
```diff
@@ -346,6 +355,22 @@ public Path getThemesDirectory() {
...
+          .sorted(Comparator.comparingLong(p -> p.toFile().lastModified()))
+          .sorted(Collections.reverseOrder())
+          .skip(NUMBER_GAME_LOGS_STORED - 1)
+          .forEach(p -> noCatch(() -> Files.delete(p)));
+    } catch (IOException e) {
+      logger.error("Could not list log directory.", e);
+    } catch (NoCatchException e) {
+      logger.error("Could not delete game log file");
+    }
+    return getFafLogDirectory().resolve(new SimpleDateFormat("'game_'yyyy-MM-dd_HHmmss'.log'").format(new Date()));
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/preferences/PreferencesService.java]: How is that gonna look are there no seperation between Hours and minutes?

**Review by 1-alex98 — DISMISSED** [review]
I like, need to test this some day

File: src/main/java/com/faforever/client/preferences/PreferencesService.java (hunk 1 above)
Comment (Sheikah45) [reply src/main/java/com/faforever/client/preferences/PreferencesService.java]: Screenshot (50)

Thought this was a standard time format but can be changed if you have one you like better that is file valid

Comment (1-alex98) [reply src/main/java/com/faforever/client/preferences/PreferencesService.java]: idk I find it hard to read. Btw the old client used the game uuid as a name. Not sure of that would be an advantage. I think that was so you could match replay and game log.

Comment (Sheikah45) [reply src/main/java/com/faforever/client/preferences/PreferencesService.java]: That might be better I will just change to that if it works for you

Comment (1-alex98) [reply src/main/java/com/faforever/client/preferences/PreferencesService.java]: Yeah sure. This is not what we gotta do, I also like the time. I am indecisive.

Comment (Sheikah45) [reply src/main/java/com/faforever/client/preferences/PreferencesService.java]: Yeah personally I think the time might be more intuitive for the user than game uid and I can add a dash separator for the time

**Comment by bukajsytlos** [comment]
sry maybe nitpicking, but could java.time API be used? I would say game id is better. date can be from FS

**Review by 1-alex98 — DISMISSED** [review]

**Comment by 1-alex98** [comment]
CI fails and i still gotta test it

**Review by 1-alex98 — APPROVED** [review]
works

**Pull request merged**